ot_filter.go

```diff
@@ -0,0 +1,50 @@
+package otrestful
+
+import (
+	"github.com/emicklei/go-restful"
+	"github.com/opentracing/opentracing-go"
+	"github.com/opentracing/opentracing-go/ext"
+)
+
+// go-restful FilterFunction does not allow to opt in third-party parameters,
+// so using global variables for the moment
+// TODO(mqliang): using parameter instead of global variable to set this
+var (
+	Tracer            opentracing.Tracer
+	OperationNameFunc func(r *restful.Request) string
+	ComponentName     string
+)
+
+var (
+	DefaulOperationNameFunc = func(r *restful.Request) string {
+		return "HTTP " + r.Request.Method
```

Comment:
Unfortunately URIs are considered bad operation names, as the operation names are indexed in tracers and the cardinality of URIs are too high. The **ideal** operation name is the route that the request maps to, with the request method as a backup. So `/users/{user_id}` is ideal, `POST` is the next best thing. For go-restful, hopefully we can extract the route and use it as the operation name.